src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,658 @@
+        {
+            CNamePendingData npd = CNamePendingData(item.second);
+            cachedNameTable.append(
+                NameTableEntry(item.first,
+                               npd.getData(),
+                               NameTableEntry::NAME_NEW,
+                               "pending firstupdate"));
+        }
+
+        // qLowerBound() and qUpperBound() require our cachedNameTable list to be sorted in asc order
+        qSort(cachedNameTable.begin(), cachedNameTable.end(), NameTableEntryLessThan());
+    }
+
+    bool findInModel(const QString &name, int *lowerIndex=nullptr, int *upperIndex=nullptr)
+    {
+        // Find name in model
+        QList<NameTableEntry>::iterator lower = qLowerBound(
+            cachedNameTable.begin(), cachedNameTable.end(), name, NameTableEntryLessThan());
+        QList<NameTableEntry>::iterator upper = qUpperBound(
+            cachedNameTable.begin(), cachedNameTable.end(), name, NameTableEntryLessThan());
+        if (lowerIndex)
+            *lowerIndex = (lower - cachedNameTable.begin());
+        if (upperIndex)
+            *upperIndex = (upper - cachedNameTable.begin());
+        return lower != upper;
+    }
+
+    void refreshName(const valtype &inName, QString *qNameStatus = nullptr)
+    {
+        LOCK(cs_main);
+
+        std::string strName = ValtypeToString(inName);
+        int lowerIndex, upperIndex;
+        bool inModel = findInModel(QString::fromStdString(strName), &lowerIndex, &upperIndex);
+        QList<NameTableEntry>::iterator lower = (cachedNameTable.begin() + lowerIndex);
+
+        // preserve previous name state if available
+        std::string strNameStatus;
+        if (qNameStatus)
+            strNameStatus = qNameStatus->toStdString();
+        else
+            strNameStatus = "";
+
+        std::string strData = "";
+        int height = NameTableEntry::NAME_NON_EXISTING;
+        if(inModel)
+        {
+            strNameStatus = lower->nameStatus.toStdString();
+            height = lower->nHeight;
+            strData = lower->value.toStdString();
+        }
+
+        // NOTE: name_show only reflects the status of the name as it is
+        // currently existig in the chain. this means that if you issue
+        // something like a name_update and issue name_show, the name_update
+        // results will not be reflected until block confirmation
+        UniValue params (UniValue::VOBJ);
+        params.pushKV ("name", strName);
+
+        JSONRPCRequest jsonRequest;
+        jsonRequest.strMethod = "name_show";
+        jsonRequest.params = params;
+        jsonRequest.fHelp = false;
+
+        UniValue res;
+        try {
+            res = tableRPC.execute(jsonRequest);
+        } catch (const UniValue& e) {
+            UniValue message = find_value(e, "message");
+            std::string errorStr = message.get_str();
+            LogPrintf ("unexpected name_show response on refreshName=%s: %s\n",
```

Comment:
This spams `debug.log` with errors while the blockchain is syncing.  I suggest checking for the error code `RPC_CLIENT_IN_INITIAL_DOWNLOAD` and not logging anything in that case.